crossbar/worker/container.py

```diff
@@ -283,72 +326,20 @@ def panic(fail, msg):
         def on_connect_success(proto):
             component = ContainerComponent(component_id, config, proto, None)
             self.components[component_id] = component
-
-            # FIXME: this is a total hack.
-            #
```

Comment:
This is still true too, but maybe it needs a bit more of a comment: basically "all" we're doing here (in both cases) is attaching an "on close" listener; the wrapper is a hack around the fact that there's no such thing as "an on-close listener" :/

Are XXX/FIXME comments a bad use of this? I personally like keeping them around as a sort of "permission" to future people that "yes, this is weird and it would be great if it could be fixed properly" .. but maybe they're just confusing?

Reply:
Ok, have inserted your comment here with a FIXME. I've not really understood "why" this is a hack or the difference between the wrapper and having an on-close listener .. I will at some point dig into it and find out, although if in the meantime there's something I could read that explains it more concisely that would be great .. :) .. in terms of FIXME, I guess there are different ways of working (?) .. for me FIXME indicates something broken or incomplete hence something I wouldn't expect to see in production code. If a feature is intentionally incomplete or needs a redesign or refactor, I either use a # TODO, or create a ticket. As this fix addresses specific brokenness, my automatic response was to remove what I thought was the associated # FIXME .. to indicate it's no longer 'broken' .. but I can adapt to fit the codebase :)